Parser/pegen/pegen.c

```diff
@@ -530,13 +540,21 @@ _PyPegen_fill_token(Parser *p)
 
     if (p->fill == p->size) {
         int newsize = p->size * 2;
-        p->tokens = PyMem_Realloc(p->tokens, newsize * sizeof(Token *));
-        if (p->tokens == NULL) {
-            PyErr_Format(PyExc_MemoryError, "Realloc tokens failed");
+        Token **new_tokens = PyMem_Realloc(p->tokens, newsize * sizeof(Token *));
+        if (new_tokens == NULL) {
+            PyMem_Free(p->tokens);
+            p->tokens = NULL;
```

Comment:
Maybe the caller already frees token when it cleans up the parser object? I didn't check how a parser is destroyed.